aptos-move/framework/supra-framework/sources/supra_governance.move

```diff
@@ -581,10 +798,77 @@ module supra_framework::supra_governance {
+    public entry fun supra_vote(
+        voter: &signer,
+        proposal_id: u64,
+        should_pass: bool,
+    ) acquires SupraApprovedExecutionHashes, SupraGovernanceEvents, SupraGovernanceConfig {
+        supra_vote_internal(voter, proposal_id, should_pass);
+    }
+
+    /// Vote on proposal with `proposal_id` and specified voting_power from `stake_pool`.
+    /// If voting_power is more than all the left voting power of `stake_pool`, use all the left voting power.
+    /// If a stake pool has already voted on a proposal before partial governance voting is enabled, the stake pool
+    /// cannot vote on the proposal even after partial governance voting is enabled.
+    fun supra_vote_internal(
+        voter: &signer,
+        proposal_id: u64,
+        should_pass: bool,
+    ) acquires SupraApprovedExecutionHashes, SupraGovernanceEvents, SupraGovernanceConfig {
+        let voter_address = signer::address_of(voter);
+
+        let supra_governance_config = borrow_global<SupraGovernanceConfig>(@supra_framework);
+        assert!(vector::contains(&supra_governance_config.voters, &signer::address_of(voter)), error::permission_denied(EACCOUNT_NOT_AUTHORIZED));
+
+        // The voter's stake needs to be locked up at least as long as the proposal's expiration.
+        let proposal_expiration = multisig_voting::get_proposal_expiration_secs<GovernanceProposal>(
+            @supra_framework,
+            proposal_id
+        );
+        assert!(timestamp::now_seconds() >= proposal_expiration, error::invalid_argument(EPROPOSAL_IS_EXPIRE));
+
+        multisig_voting::vote<GovernanceProposal>(
+            &governance_proposal::create_empty_proposal(),
+            voter,
+            @supra_framework,
+            proposal_id,
+            should_pass,
+        );
+
+        if (std::features::module_event_migration_enabled()) {
+            event::emit(
+                SupraVote {
+                    proposal_id,
+                    voter: voter_address,
+                    should_pass,
+                },
+            );
+        };
+        let events = borrow_global_mut<SupraGovernanceEvents>(@supra_framework);
+        event::emit_event<SupraVoteEvent>(
+            &mut events.vote_events,
+            SupraVoteEvent {
+                proposal_id,
+                voter: voter_address,
+                should_pass,
+            },
+        );
+
+        let proposal_state = multisig_voting::get_proposal_state<GovernanceProposal>(@supra_framework, proposal_id);
+        if (proposal_state == PROPOSAL_STATE_SUCCEEDED) {
+            add_supra_approved_script_hash(proposal_id);
+        }
+    }
+
     public entry fun add_approved_script_hash_script(proposal_id: u64) acquires ApprovedExecutionHashes {
         add_approved_script_hash(proposal_id)
     }
 
+    public entry fun add_supra_approved_script_hash_script(proposal_id: u64) acquires SupraApprovedExecutionHashes {
+        add_supra_approved_script_hash(proposal_id)
+    }
```

Comment:
Use original function. If the argument list/type is not changing, no need to rename function.